src/provider/provider_os_memory_linux.c

```diff
@@ -142,14 +149,25 @@ int os_mmap_aligned(void *hint_addr, size_t length, size_t alignment, int prot,
     if (alignment > page_size) {
         uintptr_t addr = (uintptr_t)ptr;
         uintptr_t aligned_addr = (addr + alignment) & ~(alignment - 1);
+        assert_is_page_aligned(aligned_addr, page_size);
 
         size_t head_len = aligned_addr - addr;
         if (head_len > 0) {
+            // head_len has to be page-aligned, because aligned_addr has to be page-aligned
+            assert_is_page_aligned(head_len, page_size);
```

Comment:
Why do we just crash? Assert indicates a mistake in the code logic. Is this condition only possible by mistake? what if the page size 4kb and the requested alignment 6kb?

Reply:
Code changed.
Done.

Reply:
BTW tests added in a separate PR: https://github.com/oneapi-src/unified-memory-framework/pull/93

Reply:
I still only see asserts and not the INVALID_ARGUMENT error code.

Reply:
Assert indicates a mistake in the code logic. Those conditions are possible only by mistake.

Reply:
I meant those asserts verify if `os_alloc()` passed a correct alignment. A user cannot trigger them changing the input alignment. All required checks are done in `os_alloc()` and `os_alloc()` passes only a correct alignment to this function. So IMHO those asserts are correct.

Reply:
Ah yes, sorry, I missed the PR that changed the checks in os_alloc.